### PR TITLE
DEVPROD-764 enable oom tracker by default

### PR DESCRIPTION
**Pull request by ablack12**
DEVPROD-764 
### Description
Switch oom tracker to be true by default. Will send out an email to communicate this.

### Testing
Added a unit test.

### Documentation
Updated doc

### CODE REVIEW

**Review by Kimchelly — COMMENTED** [review]
If I remember correctly, @malikchaya2 had some trouble with the OOM tracker because it was enabled in one of the agent tests and running the agent tests locally triggered the security team's alerts because the (real) OOM tracker in MacOS runs `sudo log erase --all`, which is considered suspicious activity.

Not really a perfect fix, but since we're now making it default enabled, can we either make the OOM tracker never run in tests, or change the agent tests that could trip the OOM tracker ([here](https://github.com/evergreen-ci/evergreen/blob/531efa60199309956d37d70161e8dc3c5bfbfca0/agent/agent.go#L710-L715) and [here](https://github.com/evergreen-ci/evergreen/blob/531efa60199309956d37d70161e8dc3c5bfbfca0/agent/agent.go#L909)) to use [the mock OOM tracker](https://github.com/evergreen-ci/evergreen/blob/531efa60199309956d37d70161e8dc3c5bfbfca0/agent/agent_test.go#L147)?

File: docs/Project-Configuration/Project-Configuration-Files.md [hunk 1]
```diff
@@ -959,12 +959,17 @@ task within a specific build variant).
...
-a task log message saying whether it found any OOM killed processes, with their
-PIDs. A message with PIDs will also be displayed in the metadata panel in the
-UI.
+an agent log message saying whether it found any OOM killed processes, with their
+PIDs. A message with PIDs will also be displayed in the metadata panel in the UI.
+
+To disable the OOM tracker, add the following to the top-level of your yaml.
+
+``` yaml
+- oom_tracker: false
```
Comment (Kimchelly) [inline docs/Project-Configuration/Project-Configuration-Files.md:971]: Remove `- ` since it's a top-level key rather than a list item.

Comment (ablack12) [reply docs/Project-Configuration/Project-Configuration-Files.md]: good catch thanks!

**Review by Kimchelly — APPROVED** [review]

**Comment by ablack12** [comment]
@Kimchelly I added some stuff to use the mock; unfortunately I can't think of a good way to enforce this but it seems likely to stick given that we usually copy paste or add to the existing test suite. I could add something into the code to check if we're running locally if such a thing exists? Feels hacky in a different way though

**Comment by Kimchelly** [comment]
I think if the concern arose again for new tests accidentally running the OOM tracker, we could always resort to something like checking for test-specific env var flags (such as looking for `EVGHOME`) in the future. Just shoring up the existing tests seems sufficient for now.

**Pull request merged**